### PR TITLE
Asheehan edx/dropdown autoclose fork

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
## Description

* [ ] Ensure controlled usage of `Dropdown` via `show` and `onToggle` AND uncontrolled usage of `Dropdown` (no `show` or `onToggle` props provided)
  * Possibly look into using `uncontrollable` NPM package here.
* [ ] Ensure accurate `propType` definitions.
* [ ] Add tests
* [ ] Include appropriate documentation

### Deploy Preview

Include a direct link to your changes in this PR's deploy preview here (e.g., a specific component page).

## Merge Checklist

* [ ] If your update includes visual changes, have they been reviewed by a designer? Send them a link to the Netlify deploy preview, if applicable.
* [ ] Does your change adhere to the documented [style conventions](https://github.com/openedx/paragon/blob/master/docs/decisions/0012-css-styling-conventions)?
* [ ] Do any prop types have missing descriptions in the Props API tables in the documentation site (check deploy preview)?
* [ ] Were your changes tested using all available themes (see theme switcher in the header of the deploy preview, under the "Settings" icon)?
* [ ] Is there adequate test coverage for your changes?
* [ ] Consider whether this change needs to reviewed/QA'ed for accessibility (a11y). If so, please add `wittjeff` and `adamstankiewicz` as reviewers on this PR.

## Post-merge Checklist

* [ ] Verify your changes were released to [NPM](https://www.npmjs.com/package/@edx/paragon) at the expected version.
* [ ] If you'd like, [share](https://github.com/openedx/paragon/discussions/new?category=show-and-tell) your contribution in [#show-and-tell](https://github.com/openedx/paragon/discussions/categories/show-and-tell).
* [ ] 🎉 🙌 Celebrate! Thanks for your contribution.

### CODE REVIEW

File: src/Dropdown/Dropdown.test.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    expect(screen.getByText('Dropdown Button')).toBeInTheDocument();
+    expect(screen.queryByText('foobar')).not.toBeInTheDocument();
+
+    const button = screen.getByText('Dropdown Button');
+    userEvent.click(button);
+
+    debug();
+
+    await expect(screen.queryByText('foobar')).toBeInTheDocument();
```
Comment (binodpant) [inline src/Dropdown/Dropdown.test.jsx:45]: I think you want this to be await findByText instead since the findBy are the only ones that return a promise

Comment (binodpant) [reply src/Dropdown/Dropdown.test.jsx:45]:
```
userEvent.click(document.body);
    // Why isn't the menu closing??
    debug();
    expect(screen.queryByText('foobar')).not.toBeVisible();
```
this passes but the toBeInTheDocument() fails

Comment (binodpant) [reply src/Dropdown/Dropdown.test.jsx:45]: try sth like 

```
expect(screen.getByText('Dropdown Button')).toBeInTheDocument();
    expect(screen.queryByText('foobar')).not.toBeInTheDocument();
    const button = screen.getByText('Dropdown Button');
    userEvent.click(button);
    expect(screen.queryByText('foobar')).toBeInTheDocument();
    waitFor(() => expect(screen.queryByText('foobar')).toBeVisible());
    userEvent.click(document.body);
    expect(screen.queryByText('foobar')).not.toBeVisible();
```

Comment (adamstankiewicz) [reply src/Dropdown/Dropdown.test.jsx:45]: Very similar to @binodpant's suggestion above, this a solution I landed on that seems to work:

```jsx
screen.getByText('Dropdown Button');
expect(screen.queryByText('foobar')).not.toBeInTheDocument();
expect(screen.getByText('Dropdown Button')).toBeInTheDocument();
expect(screen.queryByText('foobar')).not.toBeInTheDocument();

const button = screen.getByText('Dropdown Button');
userEvent.click(button);
await waitFor(() => expect(screen.queryByText('foobar')).toBeVisible());
userEvent.click(document.body);
await waitFor(() => expect(screen.queryByText('foobar')).not.toBeVisible());
```

Comment (alex-sheehan-edx) [reply src/Dropdown/Dropdown.test.jsx]: thanks for the feedback y'all!

**Review by binodpant — APPROVED** [review]
this now lgtm!

**Pull request merged**